src/ucp/wireup/address.c

```diff
@@ -594,6 +596,7 @@ ucs_status_t ucp_address_unpack(const void *buffer, uint64_t *remote_uuid_p,
 
             address->dev_addr   = (dev_addr_len > 0) ? dev_addr : NULL;
             address->md_index   = md_index;
+            address->dev_index  = dev_index;
```

Comment:
can't two addresses have the same dev_index?
also, where is the address->dev_index used?

Reply:
two resources in same address will have the same dev_index, iff they are on same device
in this PR it's used only for printing in ucx_info, in next PRs it will be used to selecting multi-rail transports (as commented in the commit message)

Reply:
but here when unpacking, each address gets a diffenrent dev_index

Reply:
why? dev_index is incremented only when moving to the next device in the address

Reply:
right. missed this.